client/galaxy/scripts/galaxy.js

```diff
@@ -293,10 +292,75 @@ GalaxyApp.prototype.toString = function toString() {
     return `GalaxyApp(${userEmail})`;
 };
 
-// This is not great.  Rework to be a singleton-style Galaxy app accessible/created by import
-window.Galaxy = window.Galaxy || new GalaxyApp();
 
-// ============================================================================
-export default {
-    GalaxyApp: GalaxyApp
-};
+
+/**
+ * Galaxy app singleton for use in modularized code.
+ */
+
+let instance = null;
+
+export function getGalaxyInstance() {
+    
+    // TODO: remove once iframe code is safely destroyed
+    try {
+        if (window.top !== window) {
+            if ("bundleEntries" in window.top) {
+                return window.top.bundleEntries.getGalaxyInstance();
+            }
+        }
+    } catch(err) {
+        console.warn("Error retrieving application instance for horrible iframe scripting", err);
+    }
+
+    if (instance === null) {
+        throw new Error("Uninitialized singleton galaxy");
+    }
+
+    return instance;
+}
+
+export function setGalaxyInstance(factory) {
+
+    if (instance !== null) {
+        throw new Error("Galaxy already initialized");
+    }
+
+    if (!(factory instanceof Function)) {
+        console.warn("GalaxyApp.setGalaxyInstance: Singleton factory parameter should be a function");
+        factory = (app) => new app();
+    }
+
+    console.log("Initializing galaxy instance");
+    instance = factory(GalaxyApp);
+
+    return getGalaxyInstance();
+}
+
+
+
+// Install temporary code to detect when window.Global is being accessed
+// We'll use this to hunt down all the global Galaxy references and replace
+// them with getGalaxyInstance()
+
+Object.defineProperty(window, 'Galaxy', {
+    enumerable: true,
+    get() {
```

Comment:
Clever - thanks for teaching me something! 😃

Reply:
If you like that, check out ES6 Proxies. :)